src/mscorlib/src/System/String.Manipulation.cs

```diff
@@ -541,6 +541,184 @@ public String Insert(int startIndex, String value)
             }
             return result;
         }
+
+        public static string Join(char separator, params string[] value)
+        {
+            if (value == null)
+            {
+                throw new ArgumentNullException("value");
+            }
+
+            return Join(separator, value, 0, value.Length);
+        }
+
+        public static string Join(char separator, params object[] values)
+        {
+            if (values == null)
+            {
+                throw new ArgumentNullException("values");
+            }
+
+            if (values.Length == 0)
```

Comment:
You can still share the code - just need a bool argument on what the behavior in this case should be.